airbyte-integrations/connectors/destination-kafka/src/main/resources/spec.json

```diff
@@ -0,0 +1,223 @@
+      "bootstrap_servers",
+      "topic_pattern",
+      "security_protocol",
+      "acks",
+      "transaction_timeout_ms",
+      "enable_idempotence",
+      "compression_type",
+      "batch_size",
+      "linger_ms",
+      "max_in_flight_requests_per_connection",
+      "client_dns_lookup",
+      "buffer_memory",
+      "max_request_size",
+      "retries",
+      "socket_connection_setup_timeout_ms",
+      "socket_connection_setup_timeout_max_ms",
+      "delivery_timeout_ms",
+      "send_buffer_bytes",
+      "receive_buffer_bytes"
+    ],
+    "additionalProperties": false,
+    "properties": {
+      "bootstrap_servers": {
+        "title": "Bootstrap servers",
+        "description": "A list of host/port pairs to use for establishing the initial connection to the Kafka cluster. The client will make use of all servers irrespective of which servers are specified here for bootstrapping&mdash;this list only impacts the initial hosts used to discover the full set of servers. This list should be in the form <code>host1:port1,host2:port2,...</code>. Since these servers are just used for the initial connection to discover the full cluster membership (which may change dynamically), this list need not contain the full set of servers (you may want more than one, though, in case a server is down).",
+        "type": "string",
+        "examples": ["kafka-broker1:9092,kafka-broker2:9092"],
+        "order": 0
+      },
+      "topic_pattern": {
+        "title": "Topic pattern",
+        "description": "Topic pattern in which the records will be sent. You can use patterns like '{namespace}' and/or '{stream}' to send the message to a specific topic based on these values. Notice that the topic name will be transformed to a standard naming convention.",
+        "type": "string",
+        "examples": ["sample.topic", "{namespace}.{stream}.sample"],
+        "order": 1
+      },
+      "test_topic": {
+        "title": "Test topic",
+        "description": "Topic to test if Airbyte can produce messages.",
+        "type": "string",
+        "examples": ["test.topic"],
+        "order": 2
+      },
+      "sync_producer": {
+        "title": "Sync producer",
+        "description": "Wait synchronously until the record has been sent to Kafka.",
+        "type": "boolean",
+        "default": false,
+        "order": 3
+      },
+      "security_protocol": {
+        "title": "Security protocol",
+        "description": "Protocol used to communicate with brokers.",
+        "type": "string",
+        "default": "PLAINTEXT",
+        "enum": ["PLAINTEXT", "SASL_PLAINTEXT", "SSL", "SASL_SSL"],
+        "order": 4
+      },
+      "sasl_jaas_config": {
+        "title": "SASL JAAS config",
+        "description": "JAAS login context parameters for SASL connections in the format used by JAAS configuration files.",
+        "type": "string",
+        "default": "",
+        "airbyte_secret": true,
+        "order": 5
+      },
+      "sasl_mechanism": {
```

Comment:
is this related to `security_protocol`? if so why don't we do the `oneof` suggestion like above?

Reply:
Done!